vignettes/graph_gof_showcase.Rmd

```diff
@@ -16,20 +16,38 @@ This notebook provides a brief introduction to the early version of the **GoF**
 
 # Setup
 
-## Graph Approach
-
-The graph approach uses Tsantalis [pattern4.jar](https://users.encs.concordia.ca/~nikolaos/pattern_detection.html). As with other third party tools, you should specify the path to the jar in the `tools.yml` and provide the necessary parameters of the project configuration file. 
-
 ```{r}
 rm(list = ls())
 seed <- 1
 set.seed(seed)
 ```
 
-# Project Configuration File
+
+```{r warning=FALSE,message=FALSE}
+require(kaiaulu)
+require(data.table)
+require(yaml)
+require(stringi)
+require(knitr)
+require(reticulate)
+require(magrittr)
+require(gt)
+```
+
+# The Graph Approach
+
+The graph approach uses Tsantalis [pattern4.jar](https://users.encs.concordia.ca/~nikolaos/pattern_detection.html). As with other third party tools, you should specify the path to the jar in the `tools.yml` and provide the necessary parameters of the project configuration file. 
+
+// PROPOSED EDIT: Leave The Graph Approach section as an explanation of the approach rather than delving into configuring the tools.yml. If it only serves as to point to where they should be getting the jar, maybe a reworking of this section would be necessary- while configuring tools.yml to point to their installed jar should be included in the next section. 
+
+# Project Configuration Files
```

Comment:
I think the setup could be broken into two separate sections:
1. Preparing the Java Project: Java installation and SDK config, getting the project, step-by-step compilation. How can the user determine whether the project uses Maven or Gradle? If that changes compilation steps, how so? You mention you used intelliJ, but the user may need help with doing this as well. If you choose to walk them through compiling using intelliJ, I would include screenshots and links. I would also, at least briefly, mention command line compilation options since everyone has command line. Since this is meant to teach the user how to use the tool, consider adding links to one or two pre-compiled projects. You could then use these projects when walksing through examples. This will also help the user to know whether any problems that arise are in the compilation or somewhere else.
2. Configuration: what you mentioned below about configuration, also how the user should find the paths they need

Reply:
Suggestion here is to take a quick look on apache java projects (ask @beydlern ) if they have any guidance on how to compile. IntelliJ may be the hard ad-hoc route. E.g. here: https://github.com/spring-projects/spring-framework?tab=readme-ov-file#build-from-source 

I took a quick look and says Gradle is used and to type ./gradlew build. Gradle website, in turn, says it is a open source tool to compile java: https://gradle.org/. 


Apache Tomcat is another Java project. A quick inspection on the README leads me to this: https://github.com/apache/tomcat/blob/main/BUILDING.txt . Now the project tells me rather than Gradle, it needs Apache Ant. Quick look on Google for Apache Ant is that it is also a tool to build Apache Java projects: https://ant.apache.org/ 

Pick a few more, and you will have a decent idea on what is out there in the wild to compile java projects (again we are more interested in Apache projects). 

The README at the very least should brief the user that there are a couple different ways to compile them and provide some pointers. If most projects use Gradle, then we could check if the instructions for the projects are similar or all over the place. 

In R all you have to do is type "install.packages()", maybe gradle projects are easy enough we could recommend users of this tool to use it. Brute forcing through the setup will not help the user on the decision making process nor decrease the learning curve.